src/serac/physics/utilities/variational_form/weak_form.hpp

```diff
@@ -0,0 +1,514 @@
+#pragma once
```

Comment:
Does the directory "variational_form" still make sense? I vaguely remember a conversation that this was renamed to weak form ...

Reply:
I think the plan was to rename things to `functional` at some point, which will require a bit of find/replace-ing. I'm not sure if people would prefer we make that change now or later.

Reply:
I think it's OK for now, and we can have a follow on PR to move/rename to `Functional`.

Reply:
As long as it is done quickly afterwards, I don't want this to bleed too much into users.

Reply:
I'll add an issue.